knowledge-base/merge-pdfs-with-annotations.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Merge PDF files while preserving their annotations
+description:  Merge several PDF documents while preserving their annotations in the result document using PdfStreamWriter and PdfFileSource.
+type: how-to 
+page_title:  Merge PDF files while preserving their annotations
+slug: merge-pdf-files-while-preserving-their-annotations
+position: 0
+tags: merge, pdf, annotations, PdfProcessing,  
+res_type: kb
+---
+
+|Product Version|Product|Author|
+|----|----|----|
+|2023.2.713|PdfProcessing|[Yoan Karamanov](https://www.telerik.com/blogs/author/yoan-karamanov)|
+
+## Description
+ 
+This article describes how to merge PDF documents without loss of supported annotations with the help of the [PdfStreamWriter](https://docs.telerik.com/devtools/document-processing/libraries/radpdfprocessing/formats-and-conversion/pdf/pdfstreamwriter/pdfstreamwriter?_gl=1*13soitb*_ga*MTMwNDAxMDM0LjE2NjkwMTIyMDM.*_ga_9JSNBCSF54*MTY4OTg0MjQ3Ny4yNDcuMS4xNjg5ODQ1NTU1LjQ0LjAuMA..#overview) and [PdfFileSource](https://docs.telerik.com/devtools/document-processing/libraries/radpdfprocessing/formats-and-conversion/pdf/pdfstreamwriter/pdffilesource?_gl=1*18e46d3*_ga*MTMwNDAxMDM0LjE2NjkwMTIyMDM.*_ga_9JSNBCSF54*MTY4OTg0MjQ3Ny4yNDcuMS4xNjg5ODQ1NTk3LjIuMC4w). 
```

Comment:
Better use the slugs: 
- [PdfStreamWriter]({%slug radpdfprocessing-formats-and-conversion-pdf-pdfstreamwriter-overview%})
- [PdfFileSource]({%slug radpdfprocessing-formats-and-conversion-pdf-pdfstreamwriter-pdffilesource%})